### PR TITLE
Implement November 15th hardfork

### PR DESCRIPTION
**Pull request by cpacia**
- [x] Add activation parameters
- [x] Implement OP_CHECKDATASIG
- [x] Implement OP_CHECKDATASIGVERIFY
- [x] Update sigop accounting
- [x] Enforce minimum transaction size
- [x] Enforce push only
- [x] Enforce clean stack
- [x] Add `ScriptVerifyCheckDataSig` to standard flags after activation
- [x] Enforce canonical transaction order
- [x] Update blocktemplate generation to use canonical order
- [x] Write test for CTOR. Including reorg tests.
- [x] Update blocktemplate generation to create coinbases above the minimum size.
- [x] Add params and config option for testnet1 (nov hardfork testnet).

### CODE REVIEW

File: blockchain/chain.go [hunk 1]
```diff
@@ -1039,7 +1044,12 @@ func (b *BlockChain) reorganizeChain(detachNodes, attachNodes *list.List) error
 		// to it.  Also, provide an stxo slice so the spent txout
 		// details are generated.
 		stxos := make([]SpentTxOut, 0, countSpentOutputs(block))
-		err = view.connectTransactions(block, &stxos)
+		if n.parent != nil && b.IsMagneticAnomalyEnabled(n.parent.hash) {
```
Comment (cpacia) [inline blockchain/chain.go]: After activation this check can/will change to simply:
```go
if height > b.chainparmas.MagneticAnomalyActivationHeight 
```
So this squirrely logic isn't permanent.

**Comment by zquestz** [comment]
I just wanted to thank you for your work @cpacia. I am going to dive into this tonight. Great to see bchd a reality! I have wanted this for a long time. =)

**Comment by cpacia** [comment]
@zquestz thanks! Also could use some feedback on the UTXO cache but that needs to be rebased first. And this PR probably needs to be merged before rebasing.

**Comment by cpacia** [comment]
I believe this PR is ready to do. I've added testnet1 params (the nov 15 hardfork testnet) and it's fully syncing through the activation right up to the tip.

**Comment by zquestz** [comment]
Just read through the PR. Changes look good. Only note is we should bump the version to `0.12.0-beta2` so I can update images for people using bchd in Docker/Kubernetes. =)

LGTM

**Pull request merged**